colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -308,4 +306,9 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
       worker ! Kill(id.get, DisconnectCause.TimedOut)
     }
   }
+
+  private def failRequest(s: SourcedRequest, exception: Throwable): Unit = {
+    errors.hit(tags = hpTags + ("type" -> exception.getClass.getName.replaceAll("[^\\w]", "")))
```

Comment:
Actually I definitely recall `getSimpleName` giving us some problems.... yeah I checked our old history (before open-sourcing) and we used to use `getSimpleName` but it was actually throwing exceptions for some case classes.

In terms of a common place, I don't think there's currently anywhere natural to put it.  That's why I was a little hesitant to try moving it in this PR, since I think it's going to involve a little code re-org in the process.